src/components/Explorer/index.tsx

```diff
@@ -1,48 +1,86 @@
+'use client'
+
+import _ from 'lodash'
 import {
-  Code2,
   Cog,
   Cpu,
   FileJson,
-  Film,
   Joystick,
   MoreHorizontal,
   Terminal,
-} from 'lucide-react';
-import { OpenFilesSubMenu } from '../OpenFilesTabs/OpenFilesSubMenu';
-import { File } from './File';
-import { Folder } from './Folder';
-import { SubMenu } from './SubMenu';
+  type LucideIcon,
+} from 'lucide-react'
+import { OpenFilesSubMenu } from '../OpenFilesTabs/OpenFilesSubMenu'
+import { FolderStructure } from './FolderStructure'
+import { SubMenu } from './SubMenu'
 
 export type FileType = {
-  title: string;
-  icon: React.ReactNode;
-};
+  folder: string
+  title: string
+  icon: LucideIcon
```

Comment:
I think the `icon` propery should not be a `LucideIcon` so we can use other icon libraries in the future. The `ReactNode` is still the better option here. If you don't want to set `size={16}` in each icon in this array, you can use the `ReacElement` type instead.